90zfsbootmenu/zfsbootmenu-parse-commandline.sh

```diff
@@ -21,10 +21,31 @@ fi
 
 # Let the command line override our host id.
 # shellcheck disable=SC2034
-spl_hostid=$(getarg spl_hostid=)
+cli_spl_hostid=$( getarg spl.spl_hostid ) || cli_spl_hostid=$( getarg spl_hostid )
+if [ -n "${cli_spl_hostid}" ] ; then
+  # Tests modify the value, so preserve this for a warning
+  orig_cli_spl_hostid="${cli_spl_hostid}"
+
+  # Start empty so only valid hostids are set for future use
+  spl_hostid=
+
+  # Test for decimal
+  if (( 10#${cli_spl_hostid} )) >/dev/null 2>&1 ; then
+    spl_hostid="$( printf "%08x" "${cli_spl_hostid}" )"
+  # Test for hex. Requires 0x, if present, to be stripped
+  # The change to cli_spl_hostid isn't saved outside of the test
+  elif (( 16#${cli_spl_hostid//0x/} )) >/dev/null 2>&1 ; then
+    spl_hostid="${cli_spl_hostid//0x/}"
```

Comment:
Should this go through `printf` too to normalize things like `1234` to `00001234`?

Reply:
I'd get normalized in `write_hostid`, but it should be done early as well so that it's visually consistent throughout logging.